source/extensions/quic/server_preferred_address/datasource_server_preferred_address_config.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "envoy/extensions/quic/server_preferred_address/v3/datasource.pb.h"
+#include "envoy/extensions/quic/server_preferred_address/v3/datasource.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+#include "source/common/quic/envoy_quic_server_preferred_address_config_factory.h"
+
+namespace Envoy {
+namespace Quic {
+
+class DataSourceServerPreferredAddressConfigFactory
```

Comment:
Can we add a comment here which explains conceptually how this is different from the Fixed version?